ethereum_history_api/circuits/lib/src/receipt.nr

```diff
@@ -0,0 +1,71 @@
+use crate::misc::types::{Bytes32, Address};
+use crate::header::get_header;
+use crate::verifiers::receipt::verify_receipt;
+
+global BLOOM_FILTER_SIZE = 2048;
+global MAX_LOG_DATA_SIZE = 256; // 8 * 32 (Eth word length)
+global MAX_RECEIPT_LENGTH = 1024; // TODO: Pick a reasonable value
+global MAX_RECEIPT_TREE_DEPTH = 3;
+global RECEIPT_PROOF_LEN = 1024; // TODO: Pick a reasonable value
+
+struct Log {
+	address: Address,
+	topics: [Bytes32; 4],
+	data: [u8; MAX_LOG_DATA_SIZE]
+}
+
+struct PreByzantiumTxReceipt<LOG_NUM> {
+    state_root: Bytes32,
+    cumulative_gas_used: Field,
+    logs_bloom: [u8; BLOOM_FILTER_SIZE],
+    logs: [Log; LOG_NUM]
+}
+
+struct PostByzantiumTxReceipt<LOG_NUM> {
+    status: u1,
+    cumulative_gas_used: Field,
+    logs_bloom: [u8; BLOOM_FILTER_SIZE],
+    logs: [Log; LOG_NUM]
+}
+
+struct EIP4844BlobTxReceipt<LOG_NUM> {
+    blob_gas_used: Field,
+    blob_gas_price: Field,
+    status: u1,
+    cumulative_gas_used: Field,
+    logs_bloom: [u8; BLOOM_FILTER_SIZE],
+    logs: [Log; LOG_NUM]
+}
+
+struct TxReceipt<LOG_NUM> {
+    blob_gas_used: Field,
+    blob_gas_price: Field,
+    status: u1,
+    state_root: Bytes32,
+    cumulative_gas_used: Field,
+    logs_bloom: [u8; BLOOM_FILTER_SIZE],
+    logs: [Log; LOG_NUM]
+}
+
+struct TxReceiptProof {
+	key: [u8; MAX_RECEIPT_TREE_DEPTH],
+	value: [u8; MAX_RECEIPT_LENGTH],
+	proof: [u8; RECEIPT_PROOF_LEN],
+	depth: u64
+}
+
+pub fn get_tx_receipt<LOG_NUM>(block_number: Field, tx_idx: Field) -> TxReceipt<LOG_NUM> {
```

Comment:
Maybe on function level without `tx` i.e. consistently `get_receipt` and on a type level `TxReceipt`